### DISCUSSION TITLE
HiGHS MIP solver logs: restarting tree exploration

### DISCUSSION THREAD

**Discussion by LucaVaresi** [ERGO-Code/HiGHS#1402] [Q&A] [answered]
Hello everyone,
I have seen that after some iteration in the logs of the solver  the tree exploration seems to restart from the root. I am not understanding what is the reason behind this behavior, can someone explain how this might lead to a better solution w.r.t. to the one discovered previously?

**Comment by jajhall** [comment, marked as answer]
It leads to better performance because "knowledge" of the problem has been built up from the first exploration - not knowledge in an ML sense, of course. On the second pass, better branching decisions will be made, leading to a better stack of nodes to explore
